lib/mix/tasks/shards.ex

```diff
@@ -0,0 +1,124 @@
+defmodule Mix.Tasks.Shards do
+  @moduledoc """
+  Wraps Ecto Mix Tasks to operate on all of the shards defined in EctoSharding.Configuration.
+
+  Each wrapped function accepts the same command line arguments as it's Ecto counterpart, but the repo will be ignored
+  and set to each shard as it iterates over the list.
+
+  Migrations will be stored in the directory that is set in the repo's `priv` key.  It is recommended to set this to
+  something like `priv/shards` so as not to conflict with the main database migrations.
+"""
+
+  @doc false
+  def set_repo(repo, args) do
+    {parsed, args, other} = OptionParser.parse(args, aliases: [r: :repo])
+
+    permitted = parsed
+                |> Keyword.merge(repo: repo)
+                |> OptionParser.to_argv()
+
+    [args, permitted, other] |> List.flatten()
+  end
+
+  defmodule Create do
+    @moduledoc ""
+    use Mix.Task
+    alias Mix.Tasks.Shards
+    alias EctoSharding.Configuration, as: Config
+
+    @doc "Creates the Shard DB's"
+    def run(args \\ []) do
+      Enum.each(Config.shard_repos(), fn({_name, repo}) ->
+        Shards.set_repo(repo, args)
+        |> Mix.Tasks.Ecto.Create.run()
+      end)
+    end
+  end
+
+  defmodule Drop do
+    use Mix.Task
+    alias Mix.Tasks.Shards
+    alias EctoSharding.Configuration, as: Config
+
+    @doc "Drops the Shard DB's"
+    def run(args \\ []) do
+      Enum.each(Config.shard_repos(), fn({_name, repo}) ->
+        Shards.set_repo(repo, args)
+        |> Mix.Tasks.Ecto.Drop.run()
+      end)
+    end
+  end
+
+  defmodule Migrate do
+    use Mix.Task
+    alias Mix.Tasks.Shards
+    alias EctoSharding.Configuration, as: Config
+
+    @doc "Migrates the Shard DB's"
+    def run(args \\ []) do
+      Enum.each(Config.shard_repos(), fn({_name, repo}) ->
+        Shards.set_repo(repo, args)
+        |> Mix.Tasks.Ecto.Migrate.run()
+      end)
+    end
+  end
+
+  defmodule Rollback do
+    use Mix.Task
+    alias Mix.Tasks.Shards
+    alias EctoSharding.Configuration, as: Config
+
+    @doc "Rolls back the Shard DB's"
+    def run(args \\ []) do
+      Enum.each(Config.shard_repos(), fn({_name, repo}) ->
+        Shards.set_repo(repo, args)
+        |> Mix.Tasks.Ecto.Rollback.run()
+      end)
+    end
+  end
+
+  defmodule Load do
+    use Mix.Task
+    alias Mix.Tasks.Shards
+    alias EctoSharding.Configuration, as: Config
+
+    @doc "Loads the Shard DB's from the schema dump file"
+    def run(args \\ []) do
+      Enum.each(Config.shard_repos(), fn({_name, repo}) ->
+        Shards.set_repo(repo, args)
+        |> Mix.Tasks.Ecto.Load.run()
+      end)
```

Comment:
I see two basic patterns, here: 

1. we iterate over all the shards and call `Mix.Tasks.Ecto.XXX.run()`
1. we grab the first shard and call `Mix.Tasks.Ecto.XXX.run()`

thoughts on cleaning this up a little bit? Could maybe extract a few methods to help with these? I'm happy to push up a commit to work on this, but I probably won't get to it until the weekend.

Reply:
Any suggestions on the tests?  I'm thinking maybe I need to define another set of Repos in the test config for these tests to operate on.  I think the random failures are due to the database being modified, which is not in a transaction, while the other tests are running, so part way through their db evaporates!